File: May-LeetCoding-Challange-2020/Week1/387-first_unique_character_in_a_string.cpp
class Solution {
public:
    int firstUniqChar(string s) {

        /* Time Complexity: O(n)
         * Space Complexity: O(n)
        */
        
        map<char, int> b;
        
        for(int i = 0; i < s.length(); i++){ //create map with count 
            b[s[i]]++;
        }
        
        for(int i = 0; i < s.length(); i++){ //return first char with count 1
            if(b[s[i]] == 1){
                return i;
            }
        }
        
        return -1; //no unique char found
    }
};

File: May-LeetCoding-Challange-2020/Week1/387-first-unique-character-in-a-string.py
class Solution:
    # Sol 1. Using Counter, Brute Force
    def firstUniqChar(self, s: str) -> int:
        c = collections.Counter(s)
        print(c)
        for idx, ch in enumerate(s): # Enumerate returns (count as index, character)
            if c[ch] == 1:
                return idx
        return -1
        # Time Complexity: O(n)
        # Space Complexity: O(n)
        